### PR TITLE
feat(machines): Add notification to show number of selected machines MAASENG-1355

### PR DESCRIPTION
**Pull request by ndv99**
## Done

- Added a notification to show the number of machines currently selected
- Created MachineListSelectedCount component
- Exported `getCurrentPageDisplayedMachineCount` function from `MachineListDisplayCount` to assist with [MAASENG-1688](https://warthogs.atlassian.net/browse/MAASENG-1688)

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to machine list
- Select a few machines
- Ensure a notification is displayed with the correct number of machines selected
- Click "Select all X machines"
- Ensure all machines are selected
- Click "Clear selection"
- Ensure selection is cleared
- Enter a filter into the searchbox and repeat the above steps, ensuring that "filtered" is correctly shown in the notificaiton

## Fixes

Fixes [MAASENG-1355](https://warthogs.atlassian.net/browse/MAASENG-1355)

### Some selected

image

### All selected

image

### Filtered selected

image

### All filtered selected

image

[MAASENG-1688]: https://warthogs.atlassian.net/browse/MAASENG-1688?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MAASENG-1355]: https://warthogs.atlassian.net/browse/MAASENG-1355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by petermakowski — CHANGES_REQUESTED** [review]
Columns are no longer aligned correctly (I suspect the `expanded` prop to be the culprit), see the screenshot below

Google Chrome screenshot 000210@2x

File: src/app/machines/views/MachineList/_index.scss [hunk 1]
```diff
@@ -174,4 +174,10 @@
       margin-right: $spv--x-small;
     }
   }
+
+  // .select-notification {
+  //   td.p-table__expanding-panel {
+  //     border: 0;
+  //   }
+  // }
```
[the remark this answers is not in the document]
Comment (ndv99) [reply src/app/machines/views/MachineList/_index.scss:182]: I just pushed a commit removing that :wink:

File: src/app/machines/views/MachineList/MachineListTable/MachineListTable.tsx [hunk 2]
```diff
@@ -829,6 +832,36 @@ export const MachineListTable = ({
...
+            },
+          ],
+        },
+      ];
+    } else {
+      return [];
+    }
+  }, [filter, hiddenColumns.length, machineCount, selectedCount]);
+
+  const renderRows = useMemo(
```
Comment (petermakowski) [inline src/app/machines/views/MachineList/MachineListTable/MachineListTable.tsx:860]:
```suggestion
  const machineRows = useMemo(
```
`render` implies it's a render function

File: src/app/machines/views/MachineList/MachineListTable/MachineListTable.tsx [hunk 3]
```diff
@@ -866,7 +899,7 @@ export const MachineListTable = ({
         })}
         emptyStateMsg={!machinesLoading && filter ? Label.NoResults : null}
         headers={filterColumns(headers, hiddenColumns, showActions)}
-        rows={machinesLoading ? skeletonRows : rows}
+        rows={machinesLoading ? skeletonRows : renderRows}
```
Comment (petermakowski) [inline src/app/machines/views/MachineList/MachineListTable/MachineListTable.tsx:902]:
```suggestion
        rows={machinesLoading ? skeletonRows : machineRows}
```

File: src/app/machines/views/MachineList/_index.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndv99) [reply src/app/machines/views/MachineList/_index.scss:182]: wait no i didnt i forgot to save

**Review by petermakowski — APPROVED** [review]

**Pull request merged**